main/src/ca/uwaterloo/flix/language/phase/unification/shared/Bool2.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2025 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.unification.shared
+
+/**
+  * Represents the two-valued Boolean algebra B2.
+  */
+sealed trait Bool2
+
+object Bool2 {
+  case object False extends Bool2
```

Comment:
You can't use inbuilt bools?

Reply:
We can, but it makes no difference.

Reply:
It just seems a bit unnecessary, you have to write `t1 == True || t2 == True` instead of `t1 || t2` etc.

> but it makes no difference.

I'd be impressed if thats true for scala